src/bin/bench.rs

```diff
@@ -159,9 +165,108 @@ fn main() {
 
 fn run(config: Config) -> Result<(), Box<Error>> {
     println!("running benchmarking suite...");
+
+    // the thread pool we use to perform our operations on the tree
+    let pool = rayon::ThreadPool::new(rayon::Configuration::new().num_threads(config.num_threads))
+        .unwrap();
+
+    // create a default rsdb config
+    let rsdb_config = rsdb::Config::default();
+    let tree = rsdb_config.tree();
+
+    let sum_ops = config.set + config.scan + config.get + config.delete + config.cas;
+    let ops = vec![
+        (Op::Set, config.set),
+        (Op::Scan, config.scan),
+        (Op::Get, config.get),
+        (Op::Delete, config.delete),
+        (Op::Cas, config.cas),
+    ];
+
+    let mut operations: Vec<Box<Fn() + Sync>> = Vec::new();
+    for i in 0..config.num_operations {
+        let op = get_operation_choice(&ops, sum_ops);
+
+        let operation: Box<Fn() + Sync> = match op {
+            Some(&Op::Set) => Box::new(|| perform_set_operation(&tree)),
+            Some(&Op::Scan) => Box::new(|| perform_scan_operation(&tree)),
+            Some(&Op::Get) => Box::new(|| perform_get_operation(&tree)),
+            Some(&Op::Delete) => Box::new(|| perform_delete_operation(&tree)),
+            Some(&Op::Cas) => Box::new(|| perform_cas_operation(&tree)),
+            _ => Box::new(|| {}),
```

Comment:
one thing I worry about is if we have to do billions of operations, how much memory will that take up? it may be better to generate the operation on the fly rather than creating a vector of operations to pop off for each one, and probably faster too.